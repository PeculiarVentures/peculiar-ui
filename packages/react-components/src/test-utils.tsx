import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
