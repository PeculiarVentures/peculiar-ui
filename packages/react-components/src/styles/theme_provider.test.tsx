import React from 'react';
import { render } from '../test-utils';
import { ThemeProvider } from './index';

describe('<ThemeProvider />', () => {
  it('should render with light theme', () => {
    render(
      <ThemeProvider>
        <div>
          Application
        </div>
      </ThemeProvider>,
    );

    expect(document.head).toMatchSnapshot();
  });

  it('should render with dark theme', () => {
    render(
      <ThemeProvider mode="dark">
        <div>
          Application
        </div>
      </ThemeProvider>,
    );

    expect(document.head).toMatchSnapshot();
  });
});
