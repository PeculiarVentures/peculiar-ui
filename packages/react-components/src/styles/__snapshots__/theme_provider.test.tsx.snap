// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeProvider /> should render with dark theme 1`] = `
<head>
  <style
    data-emotion="css-global"
    data-s=""
  >
    
    html,::backdrop{--pv-color-primary-tint-5:#e8ffe5;--pv-color-primary-tint-4:#c2eabe;--pv-color-primary-tint-3:#9ed698;--pv-color-primary-tint-2:#7fc178;--pv-color-primary-tint-1:#62ad5a;--pv-color-primary:#499940;--pv-color-primary-shade-1:#45823e;--pv-color-primary-shade-2:#3e6b39;--pv-color-primary-shade-3:#355431;--pv-color-primary-shade-4:#293d27;--pv-color-primary-shade-5:#1b261a;--pv-color-primary-contrast:#FAFAFA;--pv-color-secondary-tint-5:#e5eeff;--pv-color-secondary-tint-4:#ccdeff;--pv-color-secondary-tint-3:#b2ceff;--pv-color-secondary-tint-2:#99beff;--pv-color-secondary-tint-1:#7faeff;--pv-color-secondary:#67a0ff;--pv-color-secondary-shade-1:#618bd3;--pv-color-secondary-shade-2:#5775a8;--pv-color-secondary-shade-3:#485b7c;--pv-color-secondary-shade-4:#343e51;--pv-color-secondary-shade-5:#1a1e26;--pv-color-secondary-contrast:#FAFAFA;--pv-color-wrong-tint-5:#ffe6e5;--pv-color-wrong-tint-4:#fccbc9;--pv-color-wrong-tint-3:#f9afac;--pv-color-wrong-tint-2:#f7928f;--pv-color-wrong-tint-1:#f47773;--pv-color-wrong:#f55d58;--pv-color-wrong-shade-1:#c95a56;--pv-color-wrong-shade-2:#a05350;--pv-color-wrong-shade-3:#774644;--pv-color-wrong-shade-4:#4f3332;--pv-color-wrong-shade-5:#261b1a;--pv-color-wrong-contrast:#FAFAFA;--pv-color-attention-tint-5:#fff4e5;--pv-color-attention-tint-4:#ffebce;--pv-color-attention-tint-3:#ffe1b5;--pv-color-attention-tint-2:#ffd79b;--pv-color-attention-tint-1:#ffcd82;--pv-color-attention:#ffc369;--pv-color-attention-shade-1:#d3a663;--pv-color-attention-shade-2:#a88859;--pv-color-attention-shade-3:#7c6849;--pv-color-attention-shade-4:#514635;--pv-color-attention-shade-5:#26211a;--pv-color-success-tint-5:#e5fff6;--pv-color-success-tint-4:#caf9ea;--pv-color-success-tint-3:#aff7df;--pv-color-success-tint-2:#93f2d2;--pv-color-success-tint-1:#7aefc8;--pv-color-success:#61edbe;--pv-color-success-shade-1:#5cc4a1;--pv-color-success-shade-2:#529b83;--pv-color-success-shade-3:#457565;--pv-color-success-shade-4:#314c43;--pv-color-success-shade-5:#1a2622;--pv-color-white:#FAFAFA;--pv-color-gray-1:#2D3338;--pv-color-gray-2:#32383D;--pv-color-gray-3:#40474D;--pv-color-gray-4:#4F565C;--pv-color-gray-5:#5B6266;--pv-color-gray-6:#7E8388;--pv-color-gray-7:#A9ADAF;--pv-color-gray-8:#D4D6D7;--pv-color-gray-9:#E9EAEB;--pv-color-gray-10:#ECEDED;--pv-color-black:#23292B;--pv-color-extra-1:#826FF6;--pv-color-extra-2:#EC9CEE;--pv-shadow-light-low:0px 1px 7px 0px rgba(50, 56, 61, 0.06),0px 3px 3px 0px rgba(50, 56, 61, 0.08),0px 1px 3px 0px rgba(50, 56, 61, 0.12);--pv-shadow-light-medium:0px 1px 10px 0px rgba(50, 56, 61, 0.12),0px 3px 5px 0px rgba(50, 56, 61, 0.14),0px 1px 4px 0px rgba(50, 56, 61, 0.18);--pv-shadow-light-hight:0px 15px 20px 0.2px rgba(50, 56, 61, 0.21),0px 5px 10px 0.2px rgba(50, 56, 61, 0.25),0px 1px 8px -2px rgba(50, 56, 61, 0.20);--pv-shadow-light-soft:0px 15px 20px 0.2px rgba(50, 56, 61, 0.05),0px 5px 10px 0.2px rgba(50, 56, 61, 0.03),0px 1px 8px -2px rgba(50, 56, 61, 0.03);--pv-shadow-dark-medium:0px 1px 10px 0px rgba(25, 26, 28, 0.14),0px 3px 5px 0px rgba(25, 26, 28, 0.16),0px 1px 4px 0px rgba(25, 26, 28, 0.18);--pv-shadow-dark-hight:0px 15px 20px 0.2px rgba(25, 26, 28, 0.25),0px 5px 10px 0.2px rgba(25, 26, 28, 0.30),0px 1px 8px -2px rgba(25, 26, 28, 0.20);--pv-text-h1-weight:800;--pv-text-h1-size:30px;--pv-text-h1-height:40px;--pv-text-h1-spacing:0.1px;--pv-text-h2-weight:800;--pv-text-h2-size:24px;--pv-text-h2-height:35px;--pv-text-h2-spacing:0.1px;--pv-text-h3-weight:700;--pv-text-h3-size:20px;--pv-text-h3-height:30px;--pv-text-h3-spacing:0.2px;--pv-text-h4-weight:700;--pv-text-h4-size:18px;--pv-text-h4-height:28px;--pv-text-h4-spacing:0.2px;--pv-text-h5-weight:700;--pv-text-h5-size:16px;--pv-text-h5-height:25px;--pv-text-h5-spacing:0.15px;--pv-text-s1-weight:600;--pv-text-s1-size:15px;--pv-text-s1-height:25px;--pv-text-s1-spacing:0.25px;--pv-text-s2-weight:600;--pv-text-s2-size:14px;--pv-text-s2-height:22px;--pv-text-s2-spacing:0.25px;--pv-text-b1-weight:400;--pv-text-b1-size:15px;--pv-text-b1-height:25px;--pv-text-b1-spacing:0.4px;--pv-text-b2-weight:400;--pv-text-b2-size:14px;--pv-text-b2-height:22px;--pv-text-b2-spacing:0.4px;--pv-text-b3-weight:400;--pv-text-b3-size:13px;--pv-text-b3-height:20px;--pv-text-b3-spacing:0.55px;--pv-text-btn1-weight:700;--pv-text-btn1-size:14px;--pv-text-btn1-height:20px;--pv-text-btn1-spacing:0.2px;--pv-text-btn2-weight:700;--pv-text-btn2-size:13px;--pv-text-btn2-height:17px;--pv-text-btn2-spacing:0.2px;--pv-text-c1-weight:400;--pv-text-c1-size:12px;--pv-text-c1-height:17px;--pv-text-c1-spacing:0.45px;--pv-text-c2-weight:400;--pv-text-c2-size:11px;--pv-text-c2-height:15px;--pv-text-c2-spacing:0.5px;--pv-size-base-half:2.5px;--pv-size-base:5px;--pv-size-base-2:10px;--pv-size-base-3:15px;--pv-size-base-4:20px;--pv-size-base-5:25px;--pv-size-base-6:30px;--pv-size-base-7:35px;--pv-size-base-8:40px;--pv-size-base-9:45px;--pv-size-base-10:50px;--pv-size-base-11:55px;--pv-size-base-12:60px;--pv-size-base-14:70px;--pv-size-base-16:80px;}
  </style>
</head>
`;

exports[`<ThemeProvider /> should render with light theme 1`] = `
<head>
  <style
    data-emotion="css-global"
    data-s=""
  >
    
    html,::backdrop{--pv-color-primary-tint-5:#e8ffe5;--pv-color-primary-tint-4:#c6efc2;--pv-color-primary-tint-3:#a9e2a3;--pv-color-primary-tint-2:#8ed686;--pv-color-primary-tint-1:#76c96c;--pv-color-primary:#5ebc54;--pv-color-primary-shade-1:#569e4f;--pv-color-primary-shade-2:#4b7f46;--pv-color-primary-shade-3:#3e603a;--pv-color-primary-shade-4:#2d422b;--pv-color-primary-shade-5:#1b261a;--pv-color-primary-contrast:#FFFFFF;--pv-color-secondary-tint-5:#e5efff;--pv-color-secondary-tint-4:#c2d8fc;--pv-color-secondary-tint-3:#9fc2f9;--pv-color-secondary-tint-2:#7babf7;--pv-color-secondary-tint-1:#5a95f4;--pv-color-secondary:#377ff4;--pv-color-secondary-shade-1:#4276c9;--pv-color-secondary-shade-2:#4367a0;--pv-color-secondary-shade-3:#3e5477;--pv-color-secondary-shade-4:#303c4f;--pv-color-secondary-shade-5:#1a1f26;--pv-color-secondary-contrast:#FFFFFF;--pv-color-wrong-tint-5:#ffe6e5;--pv-color-wrong-tint-4:#f7c2c0;--pv-color-wrong-tint-3:#efa09e;--pv-color-wrong-tint-2:#ea827e;--pv-color-wrong-tint-1:#e2635f;--pv-color-wrong:#de4641;--pv-color-wrong-shade-1:#b74945;--pv-color-wrong-shade-2:#934644;--pv-color-wrong-shade-3:#6d3c3b;--pv-color-wrong-shade-4:#492e2d;--pv-color-wrong-shade-5:#261b1a;--pv-color-wrong-contrast:#FFFFFF;--pv-color-attention-tint-5:#fff4e5;--pv-color-attention-tint-4:#fce4bf;--pv-color-attention-tint-3:#f9d39a;--pv-color-attention-tint-2:#f9c577;--pv-color-attention-tint-1:#f7b654;--pv-color-attention:#f7a831;--pv-color-attention-shade-1:#cc923d;--pv-color-attention-shade-2:#a37c41;--pv-color-attention-shade-3:#775f3b;--pv-color-attention-shade-4:#4f422f;--pv-color-attention-shade-5:#26211a;--pv-color-success-tint-5:#e5fff6;--pv-color-success-tint-4:#bcf4e2;--pv-color-success-tint-3:#96eace;--pv-color-success-tint-2:#72e0bb;--pv-color-success-tint-1:#51d6a9;--pv-color-success:#33cc99;--pv-color-success-shade-1:#3aaa85;--pv-color-success-shade-2:#3b896f;--pv-color-success-shade-3:#366857;--pv-color-success-shade-4:#2b473e;--pv-color-success-shade-5:#1a2622;--pv-color-white:#FFFFFF;--pv-color-gray-1:#FBFCFD;--pv-color-gray-2:#F4F7FC;--pv-color-gray-3:#EFF1F6;--pv-color-gray-4:#EAEDF2;--pv-color-gray-5:#E3E5E8;--pv-color-gray-6:#D8DBDE;--pv-color-gray-7:#CDD1D5;--pv-color-gray-8:#C4C8CC;--pv-color-gray-9:#8F999E;--pv-color-gray-10:#5C666A;--pv-color-black:#293033;--pv-color-extra-1:#826FF6;--pv-color-extra-2:#DC6CDF;--pv-shadow-light-low:0px 1px 3px rgba(85, 94, 98, 0.12),0px 3px 3px rgba(85, 94, 98, 0.08),0px 1px 7px rgba(85, 94, 98, 0.06);--pv-shadow-light-medium:0px 1px 4px rgba(85, 94, 98, 0.18),0px 3px 5px rgba(85, 94, 98, 0.14),0px 1px 10px rgba(85, 94, 98, 0.12);--pv-shadow-light-hight:0px 1px 8px -2px rgba(85, 94, 98, 0.2),0px 5px 10px 0.2px rgba(85, 94, 98, 0.25),0px 15px 20px 0.2px rgba(85, 94, 98, 0.21);--pv-shadow-light-soft:0px 15px 20px 0.2px rgba(92, 102, 106, 0.05),0px 5px 10px 0.2px rgba(92, 102, 106, 0.03),0px 1px 8px -2px rgba(92, 102, 106, 0.03);--pv-shadow-dark-medium:0px 1px 4px rgba(41, 48, 51, 0.18),0px 3px 5px rgba(41, 48, 51, 0.16),0px 1px 10px rgba(41, 48, 51, 0.14);--pv-shadow-dark-hight:0px 1px 8px -2px rgba(85, 94, 98, 0.2),0px 5px 10px 0.2px rgba(85, 94, 98, 0.3),0px 15px 20px 0.2px rgba(85, 94, 98, 0.25);--pv-text-h1-weight:800;--pv-text-h1-size:30px;--pv-text-h1-height:40px;--pv-text-h1-spacing:0.1px;--pv-text-h2-weight:800;--pv-text-h2-size:24px;--pv-text-h2-height:35px;--pv-text-h2-spacing:0.1px;--pv-text-h3-weight:700;--pv-text-h3-size:20px;--pv-text-h3-height:30px;--pv-text-h3-spacing:0.2px;--pv-text-h4-weight:700;--pv-text-h4-size:18px;--pv-text-h4-height:28px;--pv-text-h4-spacing:0.2px;--pv-text-h5-weight:700;--pv-text-h5-size:16px;--pv-text-h5-height:25px;--pv-text-h5-spacing:0.15px;--pv-text-s1-weight:600;--pv-text-s1-size:15px;--pv-text-s1-height:25px;--pv-text-s1-spacing:0.25px;--pv-text-s2-weight:600;--pv-text-s2-size:14px;--pv-text-s2-height:22px;--pv-text-s2-spacing:0.25px;--pv-text-b1-weight:400;--pv-text-b1-size:15px;--pv-text-b1-height:25px;--pv-text-b1-spacing:0.4px;--pv-text-b2-weight:400;--pv-text-b2-size:14px;--pv-text-b2-height:22px;--pv-text-b2-spacing:0.4px;--pv-text-b3-weight:400;--pv-text-b3-size:13px;--pv-text-b3-height:20px;--pv-text-b3-spacing:0.55px;--pv-text-btn1-weight:700;--pv-text-btn1-size:14px;--pv-text-btn1-height:20px;--pv-text-btn1-spacing:0.2px;--pv-text-btn2-weight:700;--pv-text-btn2-size:13px;--pv-text-btn2-height:17px;--pv-text-btn2-spacing:0.2px;--pv-text-c1-weight:400;--pv-text-c1-size:12px;--pv-text-c1-height:17px;--pv-text-c1-spacing:0.45px;--pv-text-c2-weight:400;--pv-text-c2-size:11px;--pv-text-c2-height:15px;--pv-text-c2-spacing:0.5px;--pv-size-base-half:2.5px;--pv-size-base:5px;--pv-size-base-2:10px;--pv-size-base-3:15px;--pv-size-base-4:20px;--pv-size-base-5:25px;--pv-size-base-6:30px;--pv-size-base-7:35px;--pv-size-base-8:40px;--pv-size-base-9:45px;--pv-size-base-10:50px;--pv-size-base-11:55px;--pv-size-base-12:60px;--pv-size-base-14:70px;--pv-size-base-16:80px;}
  </style>
</head>
`;
